libdrgn/arch_x86_64.c

```diff
@@ -258,9 +306,20 @@ fallback_unwind_x86_64(struct drgn_program *prog,
 	err = get_registers_from_frame_pointer(prog, rbp.value, ret);
 	if (err) {
 		if (err->code == DRGN_ERROR_FAULT) {
+			struct optional_uint64 pc;
+
 			drgn_error_destroy(err);
 			err = &drgn_stop;
+
+			/*
+			 * Check if a NULL function pointer was called
+			 * and adjust registers so unwinding is possible.
+			 */
+			pc = drgn_register_state_get_pc(regs);
+			if (pc.has_value && pc.value == 0)
```

Comment:
I think it'd be better to check this NULL call case before trying to use the frame pointer. (If frame pointers are enabled, then after a call of NULL, the frame pointer will still point to the caller of the function that called NULL, and thus we'll skip the function that called NULL, which is the one we care about.)